impl/src/main/java/io/kroxylicious/testing/kafka/invm/InVMKafkaCluster.java

```diff
@@ -352,49 +365,61 @@ public synchronized void startNodes(Predicate<Integer> nodeIdPredicate) {
     @SuppressWarnings("ResultOfMethodCallIgnored")
     @Override
     public synchronized void close() throws Exception {
-        try {
-            try {
-                roleOrderedShutdown(servers, false);
-            }
-            finally {
-                if (zooServer != null) {
-                    zooServer.shutdown(true);
-                }
-            }
-        }
-        finally {
-            if (tempDirectory.toFile().exists()) {
-                try (var ps = Files.walk(tempDirectory);
-                        var s = ps
-                                .sorted(Comparator.reverseOrder())
-                                .map(Path::toFile)) {
-                    s.forEach(File::delete);
-                }
-            }
-        }
+        roleOrderedShutdown(servers)
+                .whenComplete((unused, throwable) -> {
+                    if (throwable != null) {
```

Comment:
might be good to rethrow after each stage so an error bubbles out eventually to the `get`